tests/Tests.c

```diff
@@ -104,15 +104,66 @@ void test_ecdsa(Curve curve) {
     int test_message_len = strlen(test_message);
 
     char *signature = ecdsa_sign(privkey.pemPublicKey, privkey.pemPrivateKey, test_message, test_message_len);
+    assert_not_null(signature, "Successfully created digital signature");
+
+
+    bool pre_generated_message_valid = ecdsa_verify(privkey.pemPublicKey, test_message, test_message_len, signature);
+    assert_true(pre_generated_message_valid, "Successfully validated generated digital signature");
+
+    free(signature);
+    KeyPair_free(privkey);
+}
+
+void test_depreciated_ecdsa(Curve curve) {
+    log_message("Testing ECDSA digital signature - Depreciated method");
+
+    KeyPair privkey = generate_key_pair(curve);
+    KeyPair pubkey = privkey;
+    pubkey.pemPrivateKey = NULL;
+
+    int test_message_len = strlen(test_message);
+
+    char *signature = ecdsa_sign_depr(privkey.pemPublicKey, privkey.pemPrivateKey, test_message, test_message_len);
     assert_not_null(signature, "Created digital signature");
 
-    bool valid = ecdsa_verify(privkey.pemPublicKey, test_message, test_message_len, signature);
+    bool valid = ecdsa_verify_depr(privkey.pemPublicKey, test_message, test_message_len, signature);
     assert_true(valid, "Successfully validated digital signature");
 
     KeyPair_free(privkey);
     free(signature);
 }
 
+void test_pre_generated_message_ecdsa() {
+    log_message("Testing pre generated ECDSA digital signature");
+
+    char publicKey[] = "LS0tLS1CRUdJTiBQVUJMSUMgS0VZLS0tLS0KTUU0d0VBWUhLb1pJemowQ0FRWUZLNEVFQUNFRE9nQUVSZGtIN1hHM1VRaGZIR1RzQmJ5alVXRmU2SFNycmxZWQpYcm95b0cvdGszMDlxaEprbGtCcGN0eWV2OUJIQUE0VlVPWi9GSytpNzZFPQotLS0tLUVORCBQVUJMSUMgS0VZLS0tLS0K";
+//    char privateKey[] = "LS0tLS1CRUdJTiBQUklWQVRFIEtFWS0tLS0tCk1IZ0NBUUF3RUFZSEtvWkl6ajBDQVFZRks0RUVBQ0VFWVRCZkFnRUJCQndsOUhvMDd0VTZaUW1kSGhRV01OUUUKR1N3Tm9McldmMVVvaFhkY29Ud0RPZ0FFU0RGWnFSRzBRb291TFpsV09KTFBSVlJqYUxLQXJZdldDRG94ZnRyUAppSVdQNGh6RlRNVDlhZHg5R24vcWpsNlNXWlVFVXp0REdEZz0KLS0tLS1FTkQgUFJJVkFURSBLRVktLS0tLQo=";
```

Comment:
Here's forgotten commented line of code

Reply:
It is on purpose. In the future it might be useful to have the private key that was used to sign the test message. It might help with debugging :). But I can remove it if you think it is bad practice.

Reply:
Yeah, let's keep this as a comment. In this case it serves as a documentation, not as a forgotten piece of unused code.